### DISCUSSION TITLE
Do I need to use Keys generated in the Apple Developer Console in this plugin, if yes where?

### DISCUSSION THREAD

**Discussion by sanchitgng** [lupidan/apple-signin-unity#130] [Q&A]
![image](https://user-images.githubusercontent.com/40668553/140493546-056aa417-8da2-4419-8c98-18b07afbd70c.png)
Also, during authentication, it is not going inside this function, the statement is this.firebaseAuth.SignInWithCredentialAsync(firebaseCredential).ContinueWithOnMainThread(task =>HandleSignInWithUser(task);
but somehow it is not getting executed which got me in thinking that maybe some server side validation or something is missing, please help me in this.
I have been following the [SingIn with Firebase](https://github.com/lupidan/apple-signin-unity/wiki/Sign-in-with-Apple-Unity-Plugin:-Working-with-Firebase) and also [Lupidan](https://github.com/lupidan/apple-signin-unity)

Please help me @lupidan

**Comment by imaishika** [comment]
Yes, even I am stuck with the exact issue. Please answer...

**Comment by lupidan** [comment]
This plugin objective is to integrate Apple's native login on your Unity games. The Firebase guide is a *tentative guide* adapting the [code](https://firebase.google.com/docs/auth/ios/apple#before-you-begin) that Firebase recommends on iOS to C#.
I myself haven´t tried using Firebase because I do not have any project that uses it, but I have gotten reports of people successfully using this plugin alongside Firebase. So I would say there might be something wrong with your setup.

I don't know your setup or code. In fact, you are just showing me a screenshot of a method **that´s not even called...** 🤷 
I can only make recommendations as to:
- The order of calls is first you ask Apple to login => then you pass the information to Firebase to create a user in Firebase
- Make sure you have setup the Firebase accordingly to support sign in process with your particular app
- Make sure you are sending Firebase the correct information (correct nonce, etc...)
- **Make sure** you are calling `Update` on the manager to get the callback after you successfully log in.
